### PR TITLE
chore: migrate to PDM

### PR DESCRIPTION
**Pull request by cofin**
This PR changes the packend from `poetry` to `pdm`.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -32,27 +32,20 @@ jobs:
...
-        uses: snok/install-poetry@v1
-      - name: Load cached venv
-        id: cached-poetry-dependencies
-        uses: actions/cache@v3
+      - uses: pdm-project/setup-pdm@v3
+        name: Set up PDM
         with:
-          path: .venv
-          key: venv-${{ runner.os }}-${{ matrix.python-version }}-${{ hashFiles('**/poetry.lock') }}
+          python-version: ${{ matrix.python-version }}
```
Comment (guacs) [inline .github/workflows/ci.yaml:38]: I think it's probably better to cache these as seen [here](https://github.com/litestar-org/polyfactory/blob/79360f18340da6ab67808a4a177349b206432bc9/.github/workflows/ci.yaml#L59-L64).

File: docs/developer_guide/developer_setup.md [hunk 2]
```diff
@@ -2,10 +2,10 @@
...
 
-## Using poetry
+## Using PDM
 
 ```bash
-poetry install
+pdm install
 ```
 
 Install optional dependencies using the `--extras` flag:
```
Comment (guacs) [inline docs/developer_guide/developer_setup.md:11]: It should be using `pdm` instead of `poetry.`

File: pyproject.toml [hunk 3]
```diff
@@ -237,21 +267,22 @@ classmethod-decorators = [
...
-known-first-party = ['tests', 'app']
-
-[tool.ruff.pydocstyle]
-convention = "google"
+[tool.slotscheck]
+strict-imports = false
 
 [tool.codespell]
 ignore-words-list = "alog"
+skip = 'pdm.lock, poetry.lock, package-lock.json'
```
Comment (guacs) [inline pyproject.toml:285]: `poetry.lock` can be removed right?

**Review by guacs — APPROVED** [review]

File: docs/developer_guide/developer_setup.md (hunk 2 above)
Comment (guacs) [reply docs/developer_guide/developer_setup.md:11]: By this, I meant the command below this line. GitHub doesn't let me comment on the unchanged lines, or if it does I don't know how :P

File: Makefile
Comment (JacobCoffee) [inline Makefile:1]: I think we need to add these - 

https://github.com/litestar-org/polyfactory/blob/79360f18340da6ab67808a4a177349b206432bc9/Makefile#L68-L74

I have yet to make PDM locking work myself but these always do the right thing

**Review by JacobCoffee — APPROVED** [review]

**Pull request merged**